### DISCUSSION TITLE
Replacing string input by enum

### DISCUSSION THREAD

**Discussion by Vishwa1705** [aws/aws-sdk-go-v2#1853] [General]
Hi there,

In Reserved Instance's recommendation page, We see 5 of the Services in drop down. While using sdk, i noticed that to we need to provide the service name as string input. 

Can you pls tell us how can we use enum over there instead of manually adding string for service field in GetReservationPurchaseRecommendationInput of costexplorer?

Thanks!

**Comment by RanVaknin** [comment]
Hi @c0sm1k ,

Unfortunately that is not an option because the input type is not determined by the SDK. Each AWS service determines the input types for the API models and the SDKs are auto generated based on those service models.

You can see the API spec in the [official docs](https://docs.aws.amazon.com/aws-cost-management/latest/APIReference/API_GetReservationPurchaseRecommendation.html#awscostmanagement-GetReservationPurchaseRecommendation-request-Service).

**Reply by Vishwa1705** [reply]
Ah i see. 

I was hoping team will add enums for the service field in above mentioned input as they are fixed. 

Thought it would be better to use values from sdk instead of adding them and ranging over each. 

Thanks man

**Reply by RanVaknin** [reply]
My pleasure! 🙏
